UnityProject/Assets/Scripts/Communications/SignalEmitter.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using Managers;
+using ScriptableObjects.Communications;
+using UnityEngine;
+using Mirror;
+
+namespace Communications
+{
+	public class SignalEmitter : NetworkBehaviour
```

Comment:
I don't think this class is meant to ever be used directly, right? Make it ``abstract`` so we can make sure it is never instantiated.